File: domain/src/main/java/com/arturogutierrez/openticator/interactor/DefaultSubscriber.java
package com.arturogutierrez.openticator.interactor;

import rx.Observer;

public abstract class DefaultSubscriber<T> implements Observer<T> {

  @Override
  public void onNext(T t) {
    // Do nothing by default
  }

  @Override
  public void onCompleted() {
    // Do nothing by default
  }

  @Override
  public void onError(Throwable e) {
    // Do nothing by default
  }
}

File: domain/src/main/java/com/arturogutierrez/openticator/interactor/DefaultSubscriber.kt
package com.arturogutierrez.openticator.interactor

import rx.Observer

abstract class DefaultSubscriber<T> : Observer<T> {

    override fun onNext(t: T) {
        // Do nothing by default
    }

    override fun onCompleted() {
        // Do nothing by default
    }

    override fun onError(e: Throwable) {
        // Do nothing by default
    }
}
